security/serializers.py

```diff
@@ -0,0 +1,11 @@
+import jsondatetime as json
+
+from django.core.serializers.json import DjangoJSONEncoder
+from django.core.signing import JSONSerializer as BaseJSONSerializer
+
+
+class SmartJSONSerializer(BaseJSONSerializer):
+    def dumps(self, obj):
+        return json.dumps(obj, separators=(',', ':'), cls=DjangoJSONEncoder)
+    def loads(self, data):
+        return json.loads(data)
```

Comment:
According to docs, you need to have a custom Decoder as well:

> Write your own serializer¶

> Note that unlike PickleSerializer, the JSONSerializer cannot handle arbitrary Python data types. As is often the case, there is a trade-off between convenience and security. If you wish to store more advanced data types including datetime and Decimal in JSON backed sessions, you will need to write a custom serializer (or convert such values to a JSON serializable object before storing them in request.session). While serializing these values is fairly straightforward (DjangoJSONEncoder may be helpful), writing a decoder that can reliably get back the same thing that you put in is more fragile. For example, you run the risk of returning a datetime that was actually a string that just happened to be in the same format chosen for datetimes).

> Your serializer class must implement two methods, dumps(self, obj) and loads(self, data), to serialize and deserialize the dictionary of session data, respectively.

You could try what's suggested in top answer of https://stackoverflow.com/questions/37553665/serializing-and-deserializing-datetimefield-type-object-django-app to figure out how to deserialize the datetime string back.

Whatever your implementation is, though, you'll probably want to write some tests to test it.